gnomad_qc/v5/sample_qc/assign_ancestry.py

```diff
@@ -584,22 +584,26 @@ def union_projection_scores_and_assignments(
         sample_collisions=sample_collisions,
         project="gnomad",
     )
-    v4_gen_anc = add_project_prefix_to_sample_collisions(
-        t=v4_gen_anc,
-        sample_collisions=sample_collisions,
-        project="gnomad",
-    )
 
     # Union PCA scores.
     scores_ht = v4_scores.union(projected_scores)
 
-    # Standardize v4 ancestry schema.
-    v4_gen_anc = v4_gen_anc.rename({"pop": "gen_anc"}).drop(
-        "training_sample", "evaluation_sample", "training_pop"
+    # Union genetic ancestry group assignments. Use previous genetic ancestry group assignments for gnomAD samples.
+    # The resulting table will not retain probability scores per genetic ancestry group as these are no longer available
+    # for gnomAD genome samples.
+    projected_gen_anc = projected_gen_anc.select("gen_anc")
+    meta_ht = meta_ht.annotate(projected_gen_anc=projected_gen_anc[meta_ht.key].gen_anc)
+    meta_ht = meta_ht.annotate(
+        gen_anc=hl.if_else(meta_ht.gen_anc == "oth", "remaining", meta_ht.gen_anc)
     )
+    meta_ht = meta_ht.annotate(
+        gen_anc=hl.if_else(
+            meta_ht.project == "gnomad", meta_ht.gen_anc, meta_ht.projected_gen_anc
+        )
+    )
+    meta_ht = meta_ht.select("gen_anc")
 
-    # Union genetic ancestry group assignments.
-    gen_anc_ht = projected_gen_anc.union(v4_gen_anc)
+    gen_anc_ht = projected_gen_anc.union(meta_ht)
```

Comment:
do you still need this union since the projected gen ancs get annotated onto the meta table above?